samples/dnn/mask_rcnn.cpp

```diff
@@ -0,0 +1,233 @@
+    string line;
+    while (getline(ifs, line)) classes.push_back(line);
+
+    string colorsFile = "colors.txt";
+    ifstream colorFptr(colorsFile.c_str());
+    while (getline(colorFptr, line)) {
+        char* pEnd;
+        double r, g, b;
+        r = strtod (line.c_str(), &pEnd);
+        g = strtod (pEnd, NULL);
+        b = strtod (pEnd, NULL);
+        colors.push_back(Scalar(r, g, b, 255.0));
+    }
+
+    String textGraph = "./mask_rcnn_inception_v2_coco_2018_01_28.pbtxt";
+    String modelWeights = "./mask_rcnn_inception_v2_coco_2018_01_28/frozen_inference_graph.pb";
+
+    Net net = readNetFromTensorflow(modelWeights, textGraph);
+    net.setPreferableBackend(DNN_BACKEND_OPENCV);
+    net.setPreferableTarget(DNN_TARGET_CPU);
+
+    string str, outputFile;
+    VideoCapture cap;
+    VideoWriter video;
+    Mat frame, blob;
+
+    try {
+
+        outputFile = "mask_rcnn_out_cpp.avi";
+        if (parser.has("image"))
+        {
+            str = parser.get<String>("image");
+            cout << "Image file input : " << str << endl;
+            ifstream ifile(str);
+            if (!ifile) throw("error");
+            cap.open(str);
+            str.replace(str.end()-4, str.end(), "_mask_rcnn_out.jpg");
+            outputFile = str;
+        }
+        else if (parser.has("video"))
+        {
+            str = parser.get<String>("video");
+            ifstream ifile(str);
+            if (!ifile) throw("error");
+            cap.open(str);
+            str.replace(str.end()-4, str.end(), "_mask_rcnn_out.avi");
+            outputFile = str;
+        }
+        else cap.open(parser.get<int>("device"));
+
+    }
+    catch(...) {
+        cout << "Could not open the input image/video stream" << endl;
+        return 0;
+    }
+
+    if (!parser.has("image")) {
+        video.open(outputFile, VideoWriter::fourcc('M','J','P','G'), 28, Size(cap.get(CAP_PROP_FRAME_WIDTH), cap.get(CAP_PROP_FRAME_HEIGHT)));
+    }
+
+    static const string kWinName = "Deep learning object detection in OpenCV";
+    namedWindow(kWinName, WINDOW_NORMAL);
+
+    while (waitKey(1) < 0)
+    {
+        cap >> frame;
+
+        if (frame.empty()) {
+            cout << "Done processing !!!" << endl;
+            cout << "Output file is stored as " << outputFile << endl;
+            waitKey(3000);
+            break;
+        }
+        
+        blobFromImage(frame, blob, 1.0, Size(frame.cols, frame.rows), Scalar(), true, false);
+
+        net.setInput(blob);
+
+        std::vector<String> outNames(2);
+        outNames[0] = "detection_out_final";
+        outNames[1] = "detection_masks";
+        vector<Mat> outs;
+        net.forward(outs, outNames);
+
+        postprocess(frame, outs);
+
+        vector<double> layersTimes;
+        double freq = getTickFrequency() / 1000;
+        double t = net.getPerfProfile(layersTimes) / freq;
+        string label = format("openCV with Inference - cabelo@opensuse.org : %0.0f ms", t);
```

Comment:
You can move a contact info to the comments block in the header.

Reply:
Done